src/Symfony/Component/Messenger/Bridge/AmazonSqs/Tests/Transport/ConnectionTest.php

```diff
@@ -0,0 +1,201 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Bridge\AmazonSqs\Tests\Transport;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Messenger\Bridge\AmazonSqs\Transport\Connection;
+use Symfony\Component\Messenger\Exception\TransportException;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+class ConnectionTest extends TestCase
+{
+    public function testFromInvalidDsn()
+    {
+        $this->expectException(\InvalidArgumentException::class);
+        $this->expectExceptionMessage('The given SQS DSN "sqs://" is invalid.');
+
+        Connection::fromDsn('sqs://');
+    }
+
+    public function testFromDsn()
+    {
+        $httpClient = $this->getMockBuilder(HttpClientInterface::class)->getMock();
+        $this->assertEquals(
+            new Connection(['endpoint' => 'https://localhost', 'queue_name' => 'queue'], $httpClient),
+            Connection::fromDsn('sqs://localhost/queue', [], $httpClient)
+        );
+    }
+
+    public function testFromDsnWithOptions()
+    {
+        $httpClient = $this->getMockBuilder(HttpClientInterface::class)->getMock();
+        $this->assertEquals(
+            new Connection(['endpoint' => 'https://localhost', 'account' => '213', 'queue_name' => 'queue', 'buffer_size' => 1, 'wait_time' => 5, 'auto_setup' => false], $httpClient),
+            Connection::fromDsn('sqs://localhost/213/queue', ['buffer_size' => 1, 'wait_time' => 5, 'auto_setup' => false], $httpClient)
+        );
+    }
+
+    public function testFromDsnWithQueryOptions()
+    {
+        $httpClient = $this->getMockBuilder(HttpClientInterface::class)->getMock();
+        $this->assertEquals(
+            new Connection(['endpoint' => 'https://localhost', 'account' => '213', 'queue_name' => 'queue', 'buffer_size' => 1, 'wait_time' => 5, 'auto_setup' => false], $httpClient),
+            Connection::fromDsn('sqs://localhost/213/queue?buffer_size=1&wait_time=5&auto_setup=0', [], $httpClient)
+        );
+    }
```

Comment:
Could you add a tests to prove the the endpoint regex works properly? Add some "production looking" DNS strings and show that they are okey. 
Example, use regions like: 
- eu-north-1
- us-west-1
- ap-southeast-2

Reply:
I refactored the DSN format to use something similar to the SES mailer bridge.

The new format is `sqs://default/queue_name?region=us-west-1`
with region fallback to `eu-west-1` when not provided.